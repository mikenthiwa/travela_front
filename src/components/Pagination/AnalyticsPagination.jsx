import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';

import './_analyticsPagination.scss';

class AnalyticsPagination extends PureComponent{
  handleClick = (e) => {
    const { handlePagination } = this.props;
    const { target: { id, classList } } = e;
    if (!classList.contains('disabled')){
      handlePagination(id);
    }
  }
  renderPaginationBtn (direction, previousButtonDisabled, nextButtonDisabled) {
    const { pagination: {currentPage, pageCount, prevPage } } = this.props;
    const renderButton = (text, disabled, previousButtonDisabled, nextButtonDisabled) => (
      <button
        id={text} 
        className={(previousButtonDisabled || nextButtonDisabled) ? `pg--button ${disabled && 'button-disabled'}` :`pg--button ${disabled }`}
        onClick={this.handleClick} 
        type="button">
        {text}
      </button>
    );

    switch (direction) {
    case 'Previous':
      return renderButton(direction, prevPage === 0, previousButtonDisabled);
    case 'Next':
      return renderButton(direction, currentPage === pageCount, nextButtonDisabled);
    }
  }

  render () {
    const {pagination: {currentPage, pageCount}} = this.props;
    const previousButtonDisabled = currentPage === 1 ? true: false;
    const nextButtonDisabled = currentPage >= pageCount ? true: false;
    return(
      <div className="pg--container">
        { this.renderPaginationBtn('Previous', previousButtonDisabled, nextButtonDisabled) }
        <span className="pg--text">
          { `Showing page ${currentPage} of ${pageCount} ${pageCount > 1 ? 'pages' : 'page'}` }
        </span>
        { this.renderPaginationBtn('Next', previousButtonDisabled, nextButtonDisabled) }
      </div>
    );
  }
}

AnalyticsPagination.propTypes = {
  pagination: PropTypes.object.isRequired,
  handlePagination: PropTypes.func.isRequired
};

export default AnalyticsPagination;
