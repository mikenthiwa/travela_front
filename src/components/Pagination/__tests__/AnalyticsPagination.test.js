import React from 'react';
import { shallow } from 'enzyme';
import AnalyticsPagination from '../AnalyticsPagination';

const props = {
  pagination: {
    pageCount: 2,
    currentPage: 1,
    dataCount: 5,
    limit: 3,
    nextPage: 2,
    prevPage: 0
  },
  handlePagination: jest.fn()
};
describe('Analytics Pagination', () => {
  const { pagination, handlePagination } = props;
  it('should render travel calendar', () => {
    const wrapper = shallow(<AnalyticsPagination pagination={pagination} handlePagination={handlePagination} />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should render disabled Previous button when prevPage is 0', () => {
    const { pagination, handlePagination } = props;
    pagination.prevPage = 0;
    const wrapper = shallow(<AnalyticsPagination pagination={pagination} handlePagination={handlePagination} />);
    expect(wrapper.find('#Previous').props()).toHaveProperty('className', 'pg--button button-disabled');
  });
  it('should render disabled Next button when currentPage === pageCount', () => {
    const { pagination, handlePagination } = props;
    pagination.currentPage = 2;
    const wrapper = shallow(<AnalyticsPagination pagination={pagination} handlePagination={handlePagination} />);
    expect(wrapper.find('#Next').props()).toHaveProperty('className', 'pg--button button-disabled');
  });
  it('should render Showing page 1 of 1 page when pageCount is 1', () => {
    pagination.currentPage = 1;
    pagination.pageCount = 1;
    const wrapper = shallow(<AnalyticsPagination pagination={pagination} handlePagination={handlePagination} />);
    expect(wrapper.find('.pg--text').text()).toEqual('Showing page 1 of 1 page');
  });
  describe('When prevPage > 0', () => {
    pagination.prevPage = 1;
    const wrapper = shallow(<AnalyticsPagination pagination={pagination} handlePagination={handlePagination} />);
    it('should render enabled Previous button', () => {
      expect(wrapper.find('#Previous').props()).toHaveProperty('className', 'pg--button false');
    });
    it('should call handlePagination when Previous button is clicked', () => {
      wrapper.find('#Previous').simulate('click', { target: { id: 'Previous' , classList: { contains: jest.fn(() => false )}}});
      expect(handlePagination).toHaveBeenCalledWith('Previous');
    });
  });
  describe('When nextPage < pageCount', () => {
    const wrapper = shallow(<AnalyticsPagination pagination={pagination} handlePagination={handlePagination} />);
    it('should render enabled Next button', () => {
      expect(wrapper.find('#Next').props()).toHaveProperty('className', 'pg--button false');
    });
    it('should call handlePagination when Next button is clicked', () => {
      wrapper.find('#Next').simulate('click', { target: { id: 'Next' , classList: { contains: jest.fn(() => false )}}});
      expect(handlePagination).toHaveBeenCalledWith('Next');
    });
  });
});
