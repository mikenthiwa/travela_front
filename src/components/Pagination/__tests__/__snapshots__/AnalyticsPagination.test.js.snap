// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Analytics Pagination should render travel calendar 1`] = `
<div
  className="pg--container"
>
  <button
    className="pg--button false"
    id="Previous"
    onClick={[Function]}
    type="button"
  >
    Previous
  </button>
  <span
    className="pg--text"
  >
    Showing page 1 of 2 pages
  </span>
  <button
    className="pg--button false"
    id="Next"
    onClick={[Function]}
    type="button"
  >
    Next
  </button>
</div>
`;
